clippy_lints/src/missing_doc.rs

```diff
@@ -261,6 +264,12 @@ impl<'tcx> LateLintPass<'tcx> for MissingDoc {
 
     fn check_field_def(&mut self, cx: &LateContext<'tcx>, sf: &'tcx hir::FieldDef<'_>) {
         if !sf.is_positional() {
+            // Skip checking if the field starts with underscore and allow_unused is enabled
+            if self.allow_unused && sf.ident.as_str().starts_with('_') {
+                self.prev_span = Some(sf.span);
+                return;
+            }
+
```

Comment:
You can group `if` conditions and keep only one exit point, this should be clearer since `self.prev_span` must be adjusted in all paths before leaving:

```rust
    fn check_field_def(&mut self, cx: &LateContext<'tcx>, sf: &'tcx hir::FieldDef<'_>) {
        if !sf.is_positional()
            && !(self.allow_unused && sf.ident.as_str().starts_with('_'))
            && !is_from_proc_macro(cx, sf)
        {
            let attrs = cx.tcx.hir_attrs(sf.hir_id);
            self.check_missing_docs_attrs(cx, sf.def_id, attrs, sf.span, "a", "struct field");
        }
        self.prev_span = Some(sf.span);
    }
```

Reply:
Thank you for reviewing this PR and the guidance!